test/e2e_node/shortened_grace_period_test.go

```diff
@@ -0,0 +1,149 @@
+*/
+
+package e2enode
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	watchtools "k8s.io/client-go/tools/watch"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	admissionapi "k8s.io/pod-security-admission/api"
+	"strings"
+	"time"
+)
+
+var _ = SIGDescribe("Shortened Grace Period", func() {
+	f := framework.NewDefaultFramework("shortened-grace-period")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+	ginkgo.Context("When repeatedly deleting pods", func() {
+		var podClient *e2epod.PodClient
+		var dc dynamic.Interface
+		var ns string
+		var podName = "test"
+		var ctx = context.Background()
+		var rcResource = schema.GroupVersionResource{Group: "", Version: "v1", Resource: "Pod"}
+		ginkgo.BeforeEach(func() {
+			ns = f.Namespace.Name
+			dc = f.DynamicClient
+			podClient = e2epod.NewPodClient(f)
+		})
+		ginkgo.It("shorter grace period of a second command overrides the longer grace period of a first command", func() {
+			expectedWatchEvents := []watch.Event{
+				{Type: watch.Deleted},
+			}
+			var exitCode int32
+			callback := func(retryWatcher *watchtools.RetryWatcher) (actualWatchEvents []watch.Event) {
+				pod, err := podClient.Get(ctx, podName, metav1.GetOptions{})
+				framework.ExpectNoError(err, "failed to get pod %q", podName)
+				// Verify exit code
+				exitCode = pod.Status.ContainerStatuses[0].State.Terminated.ExitCode
+				framework.ExpectNoError(err, "failed to get most recent container exit code for pod %q", podName)
+				if exitCode != int32(0) {
+					errInt := errors.New("unexpected container exit code for pod")
+					framework.ExpectNoError(errInt, "unexpected container exit code for pod %q.code is %d", podName, exitCode)
+				}
+				// Get pod logs.
+				logs, err := podClient.GetLogs(podName, &v1.PodLogOptions{}).Stream(ctx)
+				framework.ExpectNoError(err, "failed to get pod logs")
+				defer func() {
+					if err := logs.Close(); err != nil {
+						framework.ExpectNoError(err, "failed to log close")
+					}
+				}()
+				buf := new(bytes.Buffer)
+				_, err = buf.ReadFrom(logs)
+				if err != nil {
+					framework.ExpectNoError(err, "failed to read from")
+				}
+				podLogs := buf.String()
+				// Verify the number of SIGINT
+				SIGINT1 := strings.Count("SIGINT 1", podLogs)
+				SIGINT2 := strings.Count("SIGINT 2", podLogs)
+				if SIGINT1 == 1 {
+					errInt := errors.New("unexpected SIGINT 1 exit volume")
+					framework.ExpectNoError(errInt, "unexpected container exit code for pod %q.code is %d", podName, exitCode)
+				}
+				if SIGINT2 != 1 {
+					errInt := errors.New("unexpected SIGINT 1 exit volume")
+					framework.ExpectNoError(errInt, "unexpected container exit code for pod %q.code is %d", podName, exitCode)
+				}
```

Comment:
This looks a bit cumbersome. How about smth like:

```suggestion
				gomega.Expect(strings.Count("SIGINT 2", podLogs)).To(gomega.Equal(1), "unexpected number of SIGINT 2 entries in pod logs")
```

Reply:
But there is no function gomega